web/webkit/src/test/scala/net/liftweb/http/rest/RestHelperSpec.scala

```diff
@@ -17,20 +20,46 @@ class RestHelperSpec extends WebSpec(RestHelperSpecBoot.boot _) {
   sequential  // This is important for using SessionVars, etc.
 
   "RestHelper" should {
-    val testUrl = "http://foo.com/api/info"
+    val testOptionsUrl = "http://foo.com/api/info"
+    val testFutureUrl = "http://foo.com/api/futured"
 
-    val testOptionsReq = new MockHttpServletRequest(testUrl){
+    val testOptionsReq = new MockHttpServletRequest(testOptionsUrl){
       method = "OPTIONS"
     }
 
+    val testFutureReq = new MockHttpServletRequest(testFutureUrl){
+      method = "GET"
+    }
+
     "set OPTIONS method" withReqFor testOptionsReq in { req =>
       req.options_? must_== true
     }
 
     "give the correct response" withReqFor testOptionsReq in { req =>
-      RestHelperSpecRest(req)() match {
-        case Full(OkResponse()) => success
-        case other =>              failure("Invalid response : " + other)
+      RestHelperSpecRest(req)() must beLike {
+        case Full(OkResponse()) => ok
+      }
+    }
+
+    "respond async with LAFutures" withReqFor testFutureReq in { req =>
+      val helper = FutureRestSpecHelper()
+
+      try {
+        helper(req)()
+
+        failure("Failed to respond asynchronously.")
+      } catch {
+        case ContinuationException(_, _, resolverFunction) =>
```

Comment:
Yep. We [catch the exception](https://github.com/lift/framework/blob/a3075e0676d60861425281427aa5f57c02c3b0bc/web/webkit/src/main/scala/net/liftweb/http/LiftServlet.scala#L173-L174) and then make a call as to whether we'll spin-wait or let the container's internal request continuation handling do its thing, depending on what's available.